src/Promitor.Integrations.AzureMonitor/HttpPipelinePolicies/LogOutgoingRequestsPolicy.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using GuardNet;
+using Microsoft.Extensions.Logging;
+
+namespace Promitor.Integrations.AzureMonitor.HttpPipelinePolicies{
+    public class ModifyOutgoingAzureMonitorRequestsPolicy : HttpPipelinePolicy
```

Comment:
This is an insane hack needed because the SDK doesn't work like it's supposed to when you query over a time range :/ https://github.com/Azure/azure-sdk-for-net/issues/40047

Reply:
Can we add a comment to this bug and why we have it?

Thanks for working around it btw!